app/util/appUtil.js

```diff
@@ -159,33 +231,54 @@ async function runProcess(inputs, outputs) {
   return new Error('An error occurred whilst adding an item.')
 }
 
+const getItemMetadataSingle = async (tokenId, metadataKey) => {
+  const { metadata, id } = await getItem(tokenId)
+  if (id !== tokenId) throw new Error(`Id not found: ${tokenId}`)
+
+  const metadataValue = metadata[utf8ToHex(metadataKey, METADATA_KEY_LENGTH)]
+
+  if (!metadataValue) {
+    throw new Error(`No metadata with key '${metadataKey}' for token with ID: ${tokenId}`)
+  }
+  return metadataValue
+}
+
 async function getItem(tokenId) {
   let response = {}
 
   if (tokenId) {
     await api.isReady
     const item = await api.query.simpleNftModule.tokensById(tokenId)
 
-    // TODO replace...
-    response = JSON.parse(item)
+    response = item.toJSON()
```

Comment:
is this the kind of replacement this TODO was referring to?

`toJSON` from https://polkadot.js.org/docs/api/start/types.basics/

Reply:
looks sensible